poll/check.go

```diff
@@ -0,0 +1,38 @@
+package poll
+
+import (
+	"net"
+	"os"
+)
+
+// Check is a function which will be used as check for the WaitOn method.
+type Check func(t LogT) Result
+
+// FileCheck looks on filesystem and check that the path exists.
+func FileCheck(path string) Check {
+	return func(t LogT) Result {
+		_, err := os.Stat(path)
+		if err != nil && os.IsNotExist(err) {
+			t.Logf("waiting on file %s to be available", path)
+			return Continue("file %s not available", path)
+		}
+		if err != nil {
+			return Error(err)
+		}
+
+		return Success()
+	}
+}
+
+// SocketCheck try to open a connection to the address on the
+// named network.
+func SocketCheck(network, address string) Check {
```

Comment:
How about `Connection(...)` for the function name? 

If I were reading these docs I wouldn't be too sure what values to pass the function. I think the docstring should also mention something like "See net.Dial for a description of the network and address parameters.".